include/utilities/realtime/dataman/DataMan.h

```diff
@@ -0,0 +1,468 @@
+            return sizeof(uint16_t);
+        if (dtype == "int32_t")
+            return sizeof(int32_t);
+        if (dtype == "uint32_t")
+            return sizeof(uint32_t);
+        if (dtype == "int64_t")
+            return sizeof(int64_t);
+        if (dtype == "uint64_t")
+            return sizeof(uint64_t);
+        return 0;
+    }
+
+    inline json atoj(unsigned int *array)
+    {
+        json j;
+        if (array)
+        {
+            if (array[0] > 0)
+            {
+                j = {array[1]};
+                for (unsigned int i = 2; i <= array[0]; i++)
+                {
+                    j.insert(j.end(), array[i]);
+                }
+            }
+        }
+        return j;
+    }
+
+    inline string rmquote(string in) { return in.substr(1, in.length() - 2); }
+
+    inline bool isin(string a, json j)
+    {
+        for (unsigned int i = 0; i < j.size(); i++)
+        {
+            if (j[i] == a)
+                return true;
+        }
+        return false;
+    }
+
+    inline int closest(int v, json j, bool up)
+    {
+        int s = 100, k = 0, t;
+        for (unsigned int i = 0; i < j.size(); i++)
+        {
+            if (up)
+                t = j[i].get<int>() - v;
+            else
+                t = v - j[i].get<int>();
+            if (t >= 0 && t < s)
+            {
+                s = t;
+                k = i;
+            }
+        }
+        return k;
+    }
+
+    inline void check_shape(json &p_jmsg)
+    {
+        vector<size_t> varshape;
+        if (check_json(p_jmsg, {"varshape"}))
+        {
+            varshape = p_jmsg["varshape"].get<vector<size_t>>();
+        }
+        else
+        {
+            return;
+        }
+        if (p_jmsg["putshape"] == nullptr)
+        {
+            p_jmsg["putshape"] = varshape;
+        }
+        if (p_jmsg["offset"] == nullptr)
+        {
+            p_jmsg["offset"] = vector<size_t>(varshape.size(), 0);
+        }
+        p_jmsg["putbytes"] = product(p_jmsg["putshape"].get<vector<size_t>>(),
+                                     dsize(p_jmsg["dtype"].get<string>()));
+        p_jmsg["varbytes"] =
+            product(varshape, dsize(p_jmsg["dtype"].get<string>()));
+    }
+
+    inline shared_ptr<DataMan> get_man(string method)
+    {
+        string soname = "lib" + method + "man.so";
+        void *so = NULL;
+        so = dlopen(soname.c_str(), RTLD_NOW);
+        if (so)
+        {
+            shared_ptr<DataMan> (*func)() = NULL;
+            func = (shared_ptr<DataMan>(*)())dlsym(so, "getMan");
+            if (func)
+            {
+                return func();
+            }
+            else
+            {
+                logging("getMan() not found in " + soname);
+            }
+        }
+        else
+        {
+            logging("Dynamic library " + soname +
+                    " not found in LD_LIBRARY_PATH");
+        }
+        return nullptr;
+    }
+
+    string run_cmd(string cmd)
```

Comment:
It is needed because the MdtmMan uses named pipes to communicate with MDTMFTP. In this case MdtmMan needs to delete the pipes every time it is finished, which has to be done by running an external "rm" command as far as I know. However, this is only used in MdtmMan, so I can move it to the MdtmMan.cpp for the time being.